src/main/java/com/marcusslover/plus/lib/sound/Music.java

```diff
@@ -0,0 +1,229 @@
+            Task.syncDelayed(ServerUtils.getCallingPlugin(), () -> {
+                audience.stopSound(this.loop.sound);
+                if (this.tail != null) {
+                    this.tail.send(audience);
+                }
+            }, this.loopLength - (this.ticks % this.loopLength));
+        } else {
+            audience.stopSound(this.loop.sound);
+            if (this.tail != null) {
+                this.tail.send(audience);
+            }
+        }
+    }
+
+    /**
+     * Stop the music for a command sender and play the tail if it exists.
+     *
+     * @param target The command sender.
+     */
+    public void stop(CommandSender target) {
+        this.tickTask.cancel();
+        this.loopTask.cancel();
+        if (this.loops > 0) {
+            Task.syncDelayed(ServerUtils.getCallingPlugin(), () -> {
+                target.stopSound(this.loop.sound);
+                if (this.tail != null) {
+                    this.tail.send(target);
+                }
+            }, this.loopLength - (this.ticks % this.loopLength) - 1);
+        } else {
+            target.stopSound(this.loop.sound);
+            if (this.tail != null) {
+                this.tail.send(target);
+            }
+        }
+    }
+
+    /**
+     * Stop the music from the audience without playing the tail.
+     *
+     * @param audience The audience.
+     */
+    public void stopAll(Audience audience) {
+        this.tickTask.cancel();
+        this.loopTask.cancel();
+        if (this.intro != null) {
+            audience.stopSound(this.intro().sound);
+        }
+        if (this.loop != null) {
+            audience.stopSound(this.loop().sound);
+        }
+        if (this.tail != null) {
+            audience.stopSound(this.tail().sound);
+        }
+    }
+
+    /**
+     * Stop the music from the command sender without playing the tail.
+     *
+     * @param target The command sender.
+     */
+    public void stopAll(CommandSender target) {
+        this.tickTask.cancel();
+        this.loopTask.cancel();
+        if (this.intro != null) {
+            target.stopSound(this.intro().sound);
+        }
+        if (this.loop != null) {
+            target.stopSound(this.loop().sound);
+        }
+        if (this.tail != null) {
+            target.stopSound(this.tail().sound);
+        }
+    }
+
+    @Override
+    public @NotNull <T extends CommandSender> Music send(@NotNull T target) {
+        if (this.introLength != 0 && this.intro != null) {
+            this.intro.send(target);
+            loopTask = Task.syncRepeating(ServerUtils.getCallingPlugin(), () -> {
+                this.loop.send(target);
+                this.loops++;
+            }, this.introLength + 20, loopLength);
+        } else {
+            loopTask = Task.syncRepeating(ServerUtils.getCallingPlugin(), () -> {
+                this.loop.send(target);
+                this.loops++;
+            }, 1, loopLength);
+        }
+        return this;
+    }
```

Comment:
If I consider the send methods it would appear as though this object may only be used/sent a single time. If it is sent multiple times then it will lose track of other audiences. Steps should be taken to remedy the design for this class.

A) Once this class is "played" it will trigger an IllegalStateException if it is "played" a second time. Vice versa for stopping.

B) On every play/send a session is tracked for each audience/listener. And a function for ending all listening sessions or specific sessions.